openpype/pipeline/publish/lib.py

```diff
@@ -684,56 +682,21 @@ def context_plugin_should_run(plugin, context):
     return False
 
 
-def get_instance_staging_dir(instance):
-    """Unified way how staging dir is stored and created on instances.
-
-    First check if 'stagingDir' is already set in instance data.
-    In case there already is new tempdir will not be created.
-
-    It also supports `OPENPYPE_TMPDIR`, so studio can define own temp
-    shared repository per project or even per more granular context.
-    Template formatting is supported also with optional keys. Folder is
-    created in case it doesn't exists.
-
-    Available anatomy formatting keys:
-        - root[work | <root name key>]
-        - project[name | code]
+# deprecated: backward compatibility only
+# TODO: remove in the future
+def get_custom_staging_dir_info(*args, **kwargs):
+    tr_data = get_staging_dir_profile(*args, **kwargs)
 
-    Note:
-        Staging dir does not have to be necessarily in tempdir so be careful
-        about its usage.
-
-    Args:
-        instance (pyblish.lib.Instance): Instance for which we want to get
-            staging dir.
-
-    Returns:
-        str: Path to staging dir of instance.
-    """
-    staging_dir = instance.data.get('stagingDir')
-    if staging_dir:
-        return staging_dir
-
-    anatomy = instance.context.data.get("anatomy")
+    if not tr_data:
+        return None, None
 
-    # get customized tempdir path from `OPENPYPE_TMPDIR` env var
-    custom_temp_dir = tempdir.create_custom_tempdir(
-        anatomy.project_name, anatomy)
+    return tr_data["template"], tr_data["persistence"]
 
-    if custom_temp_dir:
-        staging_dir = os.path.normpath(
-            tempfile.mkdtemp(
-                prefix="pyblish_tmp_",
-                dir=custom_temp_dir
-            )
-        )
-    else:
-        staging_dir = os.path.normpath(
-            tempfile.mkdtemp(prefix="pyblish_tmp_")
-        )
-    instance.data['stagingDir'] = staging_dir
 
-    return staging_dir
+# deprecated: backward compatibility only
+# TODO: remove in the future
+def get_instance_staging_dir(instance):
+    return _get_instance_staging_dir(instance)
 
 
```

Comment:
```suggestion
```

Reply:
Are you suggesting we don't need to maintain compatibility for this function?